hazelcast/src/main/java/com/hazelcast/wan/WanReplicationPublisher.java

```diff
@@ -49,14 +91,147 @@
     void republishReplicationEvent(WanReplicationEvent wanReplicationEvent);
```

Comment:
method name raises the question: can't i do republishing by calling `publishReplicationEvent` twice?

Reply:
Maybe the method name is a bit misleading but there's a fundamental difference between `publishReplicationEvent` and `republishReplicationEvent`. `publishReplicationEvent` is supposed to be called on the source cluster on the partition owner where the entry has actually been mutated while `republishReplicationEvent` is called on the target cluster on any random member receiving a WAN batch. Republishing then might involve some stuff like finding the partition owner and forwarding the events to it to be published.

We might also rename `republishReplicationEvent` to something like `republishReceivedWanReplicationEvent` or something simpler. I don't have any ideas at the moment.

Reply:
Sure. We have to keep in mind that we actually have a config property `isRepublishingEnabled` : https://github.com/hazelcast/hazelcast/blob/c7700c26915756d3c67476b0a278fdc04eeb0f95/hazelcast/src/main/java/com/hazelcast/config/WanReplicationRef.java#L155-L161

Reply:
if it is not a big hassle, both can be changed but decision is yours.